docs/examples.md

```diff
@@ -1028,6 +1024,298 @@ jobs:
             **${{ needs.trigger.outputs.actor_handle }}** had a failure when deploying `${{ needs.trigger.outputs.ref }}` to **${{ needs.trigger.outputs.environment }}**
 ```
 
+## Multiple Jobs with GitHub Environments
+
+A detailed example using multiple jobs, [repository environments](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment), and Terraform. As mentioned in the [README](https://github.com/github/branch-deploy#about-environments-), a deployment completes when the workflow targeting that environment completes. In this example, the branch deployment action targets a separate environment than the "actual" deployment logic, which lets us control the completion of the branch deployment while being able to manage environments separately.
+
+```yaml
+name: Branch Deploy
+
+on:
+  issue_comment:
+    types:
+      - created
+
+env:
+  # These variables are scoped to the **repository**.
+  TF_VAR_image_repository: ${{ vars.IMAGE_REPOSITORY }}
+
+permissions:
+  checks: read
+  contents: write
+  deployments: write
+  packages: read
+  pull-requests: write
+
+jobs:
+  start:
+    name: Start Branch Deployment
+    runs-on: ubuntu-latest
+
+    # Only start branch deployments on pull request comments.
+    if: ${{ github.event.issue.pull_request }}
+
+    # The deployments environment is used by the branch-deploy workflow.
+    environment: deployments
+
+    # Set the outputs to be used by the rest of the workflow.
+    outputs:
+      continue: ${{ steps.branch-deploy.outputs.continue }}
+      noop: ${{ steps.branch-deploy.outputs.noop }}
+      deployment_id: ${{ steps.branch-deploy.outputs.deployment_id }}
+      environment: ${{ steps.branch-deploy.outputs.environment }}
+      ref: ${{ steps.branch-deploy.outputs.ref }}
+      comment_id: ${{ steps.branch-deploy.outputs.comment_id }}
+      initial_reaction_id: ${{ steps.branch-deploy.outputs.initial_reaction_id }}
+      actor_handle: ${{ steps.branch-deploy.outputs.actor_handle }}
+
+    steps:
+      - name: Start Branch Deployment
+        id: branch-deploy
+        uses: github/branch-deploy@v7.3.1
+        with:
+          environment: development
+          environment_targets: development,staging,production
+          skip_completing: true
+
+  # This is the "actual" deployment logic. It uses the environment specified in
+  # the branch deployment comment (e.g. `.deploy to development`).
+  deploy:
+    needs: start
+
+    name: Deploy
+    runs-on: ubuntu-latest
+
+    # Only start after the branch deployment has initialized.
+    if: ${{ needs.start.outputs.continue == 'true' }}
+
+    # Use the environment specified by the `.noop` or `.deploy` comment.
+    environment: ${{ needs.start.outputs.environment }}
```

Comment:
This is extremely clever! I love it :rocket:

Reply:
🙇🏻 Thanks!

Reply:
Being able to do this potentially changes the entire approach to using this Action for a lot of flexibility. This is wild! Thank you :star2: